sdk/log/logger.go

```diff
@@ -35,6 +34,12 @@ func newLogger(p *LoggerProvider, scope instrumentation.Scope) *logger {
 }
 
 func (l *logger) Emit(ctx context.Context, r log.Record) {
+	param := log.EnabledParameters{}
+	param.SetSeverity(r.Severity())
+	if !l.Enabled(ctx, param) {
+		return
+	}
```

Comment:
I'm not sure we need to check the `Enabled` method inside the `Emit` method. It seems like a redundant check.

Reply:
To not permit emitting logs which should be filtered out.

The caller will usually call only Logger.Emit (without calling Logger.Enabled)

Reply:
FYI, the slog also follows the same rule. https://cs.opensource.google/go/x/exp/+/701f63a6:slog/logger.go;l=232